kv/kvserver: TestStoreRangeSplitRaceUninitializedRHS failed [skipped]

kv/kvserver.TestStoreRangeSplitRaceUninitializedRHS [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/9107962?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/9107962?buildTab=artifacts#/) on master @ [11fbbca584e3bd774ebee3ee94c49ca30a096b26](https://github.com/cockroachdb/cockroach/commits/11fbbca584e3bd774ebee3ee94c49ca30a096b26):


```
=== RUN   TestStoreRangeSplitRaceUninitializedRHS
    test_log_scope.go:161: test logs captured to: /artifacts/tmp/_tmp/33e1d369c27b9c01b2b6009c561815a3/logTestStoreRangeSplitRaceUninitializedRHS2440527177
    test_log_scope.go:79: use -show-logs to present logs inline
    client_split_test.go:2172: split at key "a\x87\xf8" failed: fetching current range descriptor value: replica unavailable: (n2,s2):2 unable to serve request to r5:{/Systemtse-a\x87\xf9?} [(n1,s1):1, (n2,s2):2, next=3, gen=9]: closed timestamp: 1679039201.857084936,0 (2023-03-17 07:46:41); raft status: {"id":"2","term":43,"vote":"1","commit":613,"lead":"0","raftState":"StatePreCandidate","applied":613,"progress":{},"leadtransferee":"0"}: encountered poisoned latch /Local/Range/System/"tse"/RangeDescriptor@0,0
    panic.go:522: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/33e1d369c27b9c01b2b6009c561815a3/logTestStoreRangeSplitRaceUninitializedRHS2440527177
--- FAIL: TestStoreRangeSplitRaceUninitializedRHS (323.67s)
```
<p>Parameters: <code>TAGS=bazel,gss,deadlock</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/kv
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestStoreRangeSplitRaceUninitializedRHS.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>


Jira issue: CRDB-25542

kv/kvserver.TestStoreRangeSplitRaceUninitializedRHS [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/9122901?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/9122901?buildTab=artifacts#/) on master @ [0a72a49eb6e4ce5823b53d02125441c3da2e94f5](https://github.com/cockroachdb/cockroach/commits/0a72a49eb6e4ce5823b53d02125441c3da2e94f5):


```
=== RUN   TestStoreRangeSplitRaceUninitializedRHS
    test_log_scope.go:161: test logs captured to: /artifacts/tmp/_tmp/33e1d369c27b9c01b2b6009c561815a3/logTestStoreRangeSplitRaceUninitializedRHS1222101082
    test_log_scope.go:79: use -show-logs to present logs inline
    client_split_test.go:2172: split at key "a\x87\xf9" failed: replica unavailable: (n2,s2):2 unable to serve request to r5:{/Systemtse-a\x87\xfa?} [(n1,s1):1, (n2,s2):2, next=3, gen=8]: closed timestamp: 1679125881.690391559,0 (2023-03-18 07:51:21); raft status: {"id":"2","term":36,"vote":"1","commit":717,"lead":"0","raftState":"StatePreCandidate","applied":717,"progress":{},"leadtransferee":"0"}: encountered poisoned latch /Local/Range/System/"tse"/Transaction/"2252b810-dc66-4593-81be-45ccff818af1"@0,0
    panic.go:522: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/33e1d369c27b9c01b2b6009c561815a3/logTestStoreRangeSplitRaceUninitializedRHS1222101082
--- FAIL: TestStoreRangeSplitRaceUninitializedRHS (387.51s)
```
<p>Parameters: <code>TAGS=bazel,gss,deadlock</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestStoreRangeSplitRaceUninitializedRHS.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

kv/kvserver.TestStoreRangeSplitRaceUninitializedRHS [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/9132680?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/9132680?buildTab=artifacts#/) on master @ [53dbb86acb1d48309530181b94838faf937084d3](https://github.com/cockroachdb/cockroach/commits/53dbb86acb1d48309530181b94838faf937084d3):


```
=== RUN   TestStoreRangeSplitRaceUninitializedRHS
    test_log_scope.go:161: test logs captured to: /artifacts/tmp/_tmp/33e1d369c27b9c01b2b6009c561815a3/logTestStoreRangeSplitRaceUninitializedRHS418021715
    test_log_scope.go:79: use -show-logs to present logs inline
    client_split_test.go:2173: split at key "a\x87\xfa" failed: replica unavailable: (n2,s2):2 unable to serve request to r5:{/Systemtse-a\x87\xfb?} [(n1,s1):1, (n2,s2):2, next=3, gen=7]: closed timestamp: 1679216257.200848665,0 (2023-03-19 08:57:37); raft status: {"id":"2","term":24,"vote":"1","commit":243,"lead":"0","raftState":"StatePreCandidate","applied":243,"progress":{},"leadtransferee":"0"}: encountered poisoned latch /Local/Range/System/"tse"/Transaction/"cacf971e-9741-4c92-9759-5a5fcd8c7067"@0,0
    panic.go:540: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/33e1d369c27b9c01b2b6009c561815a3/logTestStoreRangeSplitRaceUninitializedRHS418021715
--- FAIL: TestStoreRangeSplitRaceUninitializedRHS (178.99s)
```
<p>Parameters: <code>TAGS=bazel,gss,race</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestStoreRangeSplitRaceUninitializedRHS.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

kv/kvserver.TestStoreRangeSplitRaceUninitializedRHS [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/9132679?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/9132679?buildTab=artifacts#/) on master @ [53dbb86acb1d48309530181b94838faf937084d3](https://github.com/cockroachdb/cockroach/commits/53dbb86acb1d48309530181b94838faf937084d3):


```
=== RUN   TestStoreRangeSplitRaceUninitializedRHS
    test_log_scope.go:161: test logs captured to: /artifacts/tmp/_tmp/33e1d369c27b9c01b2b6009c561815a3/logTestStoreRangeSplitRaceUninitializedRHS3929842575
    test_log_scope.go:79: use -show-logs to present logs inline
    client_split_test.go:2173: split at key "a\x87\xfa" failed: replica unavailable: (n2,s2):2 unable to serve request to r5:{/Systemtse-a\x87\xfb?} [(n1,s1):1, (n2,s2):2, next=3, gen=7]: closed timestamp: 1679214942.308604545,0 (2023-03-19 08:35:42); raft status: {"id":"2","term":45,"vote":"1","commit":796,"lead":"0","raftState":"StateFollower","applied":796,"progress":{},"leadtransferee":"0"}: encountered poisoned latch /Local/Range/System/"tse"/Transaction/"a35af954-37a8-45a1-9a49-d0bf05f0fdba"@0,0
    panic.go:522: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/33e1d369c27b9c01b2b6009c561815a3/logTestStoreRangeSplitRaceUninitializedRHS3929842575
--- FAIL: TestStoreRangeSplitRaceUninitializedRHS (292.99s)
```
<p>Parameters: <code>TAGS=bazel,gss,deadlock</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestStoreRangeSplitRaceUninitializedRHS.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

--- a/pkg/kv/kvserver/client_split_test.go
+++ b/pkg/kv/kvserver/client_split_test.go
@@ -51,6 +51,7 @@ import (
 	"github.com/cockroachdb/cockroach/pkg/storage/enginepb"
 	"github.com/cockroachdb/cockroach/pkg/testutils"
 	"github.com/cockroachdb/cockroach/pkg/testutils/serverutils"
+	"github.com/cockroachdb/cockroach/pkg/testutils/skip"
 	"github.com/cockroachdb/cockroach/pkg/testutils/sqlutils"
 	"github.com/cockroachdb/cockroach/pkg/testutils/testcluster"
 	"github.com/cockroachdb/cockroach/pkg/ts"
@@ -2026,6 +2027,11 @@ func TestStoreSplitGCHint(t *testing.T) {
 func TestStoreRangeSplitRaceUninitializedRHS(t *testing.T) {
 	defer leaktest.AfterTest(t)()
 	defer log.Scope(t).Close(t)
+
+	// The aggressive Raft timeouts in this test prevents it from maintaining
+	// quorum and making progress when stressing under race or deadlock detection.
+	skip.UnderRace(t)
+	skip.UnderDeadlock(t)
 
 	currentTrigger := make(chan *roachpb.SplitTrigger, 1)
 	var seen struct {

